### PR TITLE
Implement sort by payment date and allow sort to be called directly with a parameter

### PR DESCRIPTION
**Pull request by alalal47**
Instead of having to go through the menu.
Reinstated StudentComparators class to improve readability

### CODE REVIEW

File: src/main/java/classify/student/StudentSorter.java [hunk 1]
```diff
@@ -31,23 +35,38 @@ public static void sortByChoice(ArrayList<Student> masterStudentList, String cho
...
     /**
-     * Checks if the choice is valid (either "1" or "2").
+     * Checks if the choice is valid (either "1", "2" or "3").
      *
      * @param choice The input choice.
      * @return True if the choice is valid, false otherwise.
      */
     public static boolean isValidChoice(String choice) {
-        return ONE.equals(choice) || TWO.equals(choice);
+        return (ONE.equals(choice) || TWO.equals(choice) || THREE.equals(choice)) ||
```
Comment (Cryolian) [inline src/main/java/classify/student/StudentSorter.java:68]: choice.equals(ONE) etc might be more readable, but this is fine as well.

File: src/main/java/classify/student/StudentSorter.java [hunk 2]
```diff
@@ -102,8 +136,20 @@ private static void listStudents(ArrayList<Student> students) {
      * @param students The list of students to be listed.
      */
     private static void listStudentsWithTotalClasses(ArrayList<Student> students) {
+        int i = 1;
+        for (Student student : students) {
```
Comment (Cryolian) [inline src/main/java/classify/student/StudentSorter.java:140]: Perhaps you could remove line 139 and 143 by using the normal for loop, and make a new function in Ui class to print out the needed details.

File: src/test/java/classify/user/InputParsingTest.java [hunk 3]
```diff
@@ -90,7 +90,12 @@ public void testHelpCommand() {
                                              + System.lineSeparator() +
                 "bye                         Exits Classify"
                                              + System.lineSeparator() +
-               "sort_name                   sorts the students by name lexicographically"
+                "sort_name                   sorts the students by name lexicographically"
```
Comment (Cryolian) [inline src/test/java/classify/user/InputParsingTest.java:93]: Good formatting to make the output readable

**Review by Cryolian — APPROVED** [review]
Only some minor nitpicks. Otherwise the code looks good.

image

File: src/main/java/classify/student/StudentSorter.java [hunk 4]
```diff
@@ -31,23 +36,38 @@ public static void sortByChoice(ArrayList<Student> masterStudentList, String cho
...
+            listStudentsByName(masterStudentList);
+            break;
         case TWO:
             listStudentsByTotalClasses(masterStudentList);
             break;
+        case TOTAL_CLASSES:
+            listStudentsByTotalClasses(masterStudentList);
+            break;
+        //@@author alalal47
+        case THREE:
```
Comment (tayponghee) [inline src/main/java/classify/student/StudentSorter.java:48]: Nice way of adding cases to the pre-existing sort

File: src/main/java/classify/student/StudentSorter.java [hunk 5]
```diff
@@ -102,8 +137,20 @@ private static void listStudents(ArrayList<Student> students) {
...
+                    student.getTotalClassesAttended());
+            i ++;
+        }
+    }
+
+    private static void listStudentsWithLastPaidDate(ArrayList<Student> students) {
+        int i = 1;
         for (Student student : students) {
-            System.out.println(student.getName() + " - Total Classes Attended: " + student.getTotalClassesAttended());
+            System.out.println(i + "." + student.getName() + " - Date of last payment: " +
```
Comment (tayponghee) [inline src/main/java/classify/student/StudentSorter.java:150]: Consider removing magic literals here

**Review by tayponghee — APPROVED** [review]

**Pull request merged**